pkg/network/tracer/tracer_test.go

```diff
@@ -1680,62 +1680,111 @@ func TestHTTPSViaLibraryIntegration(t *testing.T) {
 		regexp.MustCompile(`/[^\ ]+libssl.so[^\ ]*`),
 		regexp.MustCompile(`/[^\ ]+libgnutls.so[^\ ]*`),
 	}
+
+	b := genPayload(6)
+	randName := "veth-" + string(b[:len(b)-1])
+	randNetns := "ns-" + string(b[:len(b)-1])
+	setupCmdsTmpl := `sudo ip netns add n1
+sudo ip link add vethhost type veth peer name vethn1 netns n1
+sudo ip netns exec n1 ip a a 172.20.20.20/24 dev vethn1
+sudo ip a a 172.20.20.2/24 dev vethhost
+sudo ip l s vethhost up
+sudo ip netns exec n1 ip l s vethn1 up
+sudo ip netns exec n1 ping -c 1 172.20.20.2
+`
+	defer func() {
+		out, err := exec.Command("sudo", "ip", "netns", "delete", randNetns).CombinedOutput()
+		require.NoErrorf(t, err, "failed to cleanup netns %s via: %s\n%s", randNetns, err, string(out))
+	}()
+
+	setupCmds := strings.ReplaceAll(setupCmdsTmpl, "vethhost", randName)
+	setupCmds = strings.ReplaceAll(setupCmds, "vethn1", randName)
+	setupCmds = strings.ReplaceAll(setupCmds, "n1", randNetns)
+	scanner := bufio.NewScanner(strings.NewReader(setupCmds))
+	for scanner.Scan() {
```

Comment:
Would it be useful to move the namespace setup part into the HTTPServer, and enable it based on an option? This way we can run the HTTP server in network namespaces in other tests we need as well.